From fe2c4357d753d3a5baed7e2354f3542a1e87b46d Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Wed, 12 Oct 2022 02:54:15 +0200
Subject: [PATCH 2/2] Patch out assert in g_io_unix_get_flags

We don't implement the neccesary bits to fully support this, so in case of doubt, return readable and writeable.

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 glib/giounix.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/glib/giounix.c b/glib/giounix.c
index ad3aba2..e0237fc 100644
--- a/glib/giounix.c
+++ b/glib/giounix.c
@@ -436,7 +436,10 @@ g_io_unix_get_flags (GIOChannel *channel)
         channel->is_writeable = TRUE;
         break;
       default:
-        g_assert_not_reached ();
+	channel->is_readable = TRUE;
+        channel->is_writeable = TRUE;
+        break;
+        //g_assert_not_reached ();
     }
 
   return flags;
-- 
2.37.2
